waltz-tools/src/main/java/com/wepay/waltz/tools/client/ClientCli.java

```diff
@@ -522,6 +544,172 @@ protected String getUsage() {
+
+            options.addOption(txnPerClientOption);
+            options.addOption(intervalOption);
+            options.addOption(cfgPathOption);
+            options.addOption(numActivePartitionOption);
+        }
+
+        @Override
+        protected void processCmd(CommandLine cmd) throws SubCommandFailedException {
+            try {
+                // check required arguments
+                int numOfProducedTransactions = Integer.parseInt(cmd.getOptionValue("num-produced-txn"));
+                if (numOfProducedTransactions < 0) {
+                    throw new IllegalArgumentException("Found negative: num-produced-txn must be greater or equals to 0");
+                }
+                int avgInterval = Integer.parseInt(cmd.getOptionValue("interval"));
+                if (avgInterval < 0) {
+                    throw new IllegalArgumentException("Found negative: interval must be greater or equals to 0");
+                }
+                String configFilePath = cmd.getOptionValue("cli-config-path");
+                WaltzClientConfig waltzClientConfig = getWaltzClientConfig(configFilePath);
+
+                // check optional arguments
+                int numActivePartitions = cmd.hasOption("num-active-partitions") ? Integer.parseInt(cmd.getOptionValue("num-active-partitions")) : DEFAULT_NUMBER_ACTIVE_PARTITIONS;
+                if (numActivePartitions < 1) {
+                    throw new IllegalArgumentException("num-active-partitions must be greater or equals to 1");
+                }
+
+                allProducerReady = new CountDownLatch(1);
+                allProducerDone = new CountDownLatch(1);
+
+                produceTransactions(numActivePartitions, 1, numOfProducedTransactions, avgInterval, waltzClientConfig);
+
+                checkUncaughtExceptions();
+            } catch (Exception e) {
+                throw new SubCommandFailedException(String.format("Transaction validation failed: %s", e.getMessage()));
+            }
+        }
+
+        @Override
+        protected String getUsage() {
+            return buildUsage(NAME, DESCRIPTION, getOptions());
+        }
+    }
+
+    /**
+     * Use {@code Consumer} command to create a consumer client that
+     * reads transactions submitted to Waltz server for validation,
+     * including transaction data and optimistic locking.
+     *
+     * Consumer process is closed once total number of processed transactions
+     * reaches num-callbacks.
+     */
+    private static final class CreateConsumer extends ClientCliBaseClass {
+        private static final String NAME = "create-consumer";
+        private static final String DESCRIPTION = "Creates one consumer";
+
+        private CreateConsumer(String[] args) {
+            super(args);
+        }
+
+        @Override
+        protected void configureOptions(Options options) {
+            Option txnPerClientOption = Option.builder("nc")
+                    .longOpt("num-callbacks")
```

Comment:
nit: `txn-per-client` instead of `num-callbacks` and description can be `Specify number of transactions to be consumed`